src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final ColorSupplier color = new ColorSupplier();
```

Comment:
Using static for the 'color' field is not recommended. Static methods and fields are generally considered bad practice when they can be avoided. Instead, you can create an instance of ColorSupplier in the method where it's needed.